src/core/view/overlays/ShapeToolView.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Xournal++
+ *
+ * View active shape tools (rectangle, ellipse and so on)
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+#pragma once
+
+#include <memory>
+
+#include "util/DispatchPool.h"
+#include "view/Repaintable.h"
+
+#include "OverlayView.h"
+
+class BaseShapeHandler;
+class Range;
+class Stroke;
+
+namespace xoj::view {
+struct Context;
+class StrokeView;
+
+class ShapeToolView: public OverlayView, public Repaintable, public xoj::util::Listener<ShapeToolView> {
+
+public:
+    ShapeToolView(BaseShapeHandler* toolHandler, const Repaintable* parent);
+    virtual ~ShapeToolView() noexcept;
```

Comment:
Use override instead of virtual. Also, can this class be defined final?
```suggestion
    ~ShapeToolView() override noexcept;
```